src/main/java/com/intuit/graphql/orchestrator/batch/GraphQLServiceBatchLoader.java

```diff
@@ -140,7 +139,7 @@ private BatchFieldAuthorization getDefaultOrCustomFieldAuthorization(GraphQLCont
 
       Map<String, FragmentDefinition> svcFragmentDefinitions = filterFragmentDefinitionByService(
           key.getFragmentsByName());
-      if (serviceMetadata.hasFieldResolverDirective()) {
+      if (serviceMetadata.hasFieldResolverDirective() || serviceMetadata.isFederationService()) {
```

Comment:
same `serviceMetadata.shouldRemoveExternalFields()`

Reply:
Should be in the next external/refactor pr